src/dlgIRC.cpp

```diff
@@ -29,22 +31,49 @@
 #include <QDesktopServices>
 #include <QDir>
 #include <QScrollBar>
+#include <QShortcut>
 #include <QString>
+#include <QTextDocument>
 #include <QTime>
 #include "post_guard.h"
 
 
 dlgIRC::dlgIRC()
 {
+    mInputHistoryMax = 8;
+
     setupUi(this);
-    connection = new IrcConnection(this);
-    irc->setOpenExternalLinks(true);
-    setUnifiedTitleAndToolBarOnMac(true);
-    connect(irc, SIGNAL(anchorClicked(QUrl)), this, SLOT(anchorClicked(QUrl)));
-    connect(connection, SIGNAL(messageReceived(IrcMessage*)), this, SLOT(onMessageReceived(IrcMessage*)));
-    connect(connection, SIGNAL(connected()), this, SLOT(onConnected()));
-    connect(lineEdit, SIGNAL(returnPressed()), this, SLOT(sendMsg()));
+    setupCommandParser();
 
+    connection = new IrcConnection(this);
+    connection->setReconnectDelay(5);
+
+    lineEdit->setAttribute(Qt::WA_MacShowFocusRect, false);
```

Comment:
I don't think it's needed.   It was copy pasta from the simple client example that got left as I wasn't sure if it was needed for some reason due to the focus policy that is set on the text message display and the input field.